source/tools/glkernel-cli/PngExporter.cpp

```diff
@@ -0,0 +1,181 @@
+#include "PngExporter.h"
+
+#include "helper.h"
+
+#include <cppassist/logging/logging.h>
+
+void PngExporter::exportKernel() {
+    png_bytepp pngData;
+    int colorType;
+    png_uint_32 width;
+    png_uint_32 height;
+    if (m_kernel.hasType<glkernel::kernel4>())
+    {
+        colorType = PNG_COLOR_TYPE_RGBA;
+        const auto kernel = m_kernel.value<glkernel::kernel4>();
+        width = kernel.width();
```

Comment:
width and height could be updated outside of the if to remove duplication

Reply:
This would require changes in glkernel, as width() etc. can only be called on templated tkernels